core/services/ocr3/plugins/ccip_integration_tests/home_chain/home_chain_test.go

```diff
@@ -0,0 +1,238 @@
+package home_chain
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	ccipreader "github.com/smartcontractkit/ccipocr3/pkg/reader"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/assets"
+	capcfg "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/ccip_capability_configuration"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	logger2 "github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/p2pkey"
+	helpers "github.com/smartcontractkit/chainlink/v2/core/services/ocr3/plugins/ccip_integration_tests"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	chainA  uint64 = 1
+	fChainA uint8  = 1
+
+	chainB  uint64 = 2
+	fChainB uint8  = 2
+
+	chainC  uint64 = 3
+	fChainC uint8  = 3
+)
+
+func TestHomeChainReader(t *testing.T) {
+	const (
+		ContractName      = "CCIPCapabilityConfiguration"
+		FnGetChainConfigs = "getAllChainConfigs"
+	)
+	// Initialize chainReader
+	cfg := evmtypes.ChainReaderConfig{
+		Contracts: map[string]evmtypes.ChainContractReader{
+			ContractName: {
+				ContractABI: capcfg.CCIPCapabilityConfigurationMetaData.ABI,
+				Configs: map[string]*evmtypes.ChainReaderDefinition{
+					FnGetChainConfigs: {
+						ChainSpecificName: FnGetChainConfigs,
+					},
+				},
+			},
+		},
+	}
+	//============================Setup Backend===================================
+	transactor := testutils.MustNewSimTransactor(t)
+	backend := backends.NewSimulatedBackend(core.GenesisAlloc{
+		transactor.From: {Balance: assets.Ether(1000).ToInt()},
+	}, 30e6)
+	//==============================Setup Contracts - Add capabilities=================================
+	capRegAddress, capRegContract, err := prepareCapabilityRegistry(t, backend, transactor)
+	require.NoError(t, err)
+	capConfAddress, capConfContract, err := prepareCCIPCapabilityConfig(t, backend, transactor, capRegAddress)
+	require.NoError(t, err)
+	p2pIDS := addCapabilities(t, backend, transactor, capRegContract, capConfAddress)
+	//==============================Apply configs to Capability Contract=================================
+	chainAConf := setupConfigInfo(chainA, p2pIDS, fChainA, []byte{})
+	chainBConf := setupConfigInfo(chainB, p2pIDS[1:], fChainB, []byte{})
+	chainCConf := setupConfigInfo(chainC, p2pIDS[2:], fChainC, []byte{})
```

Comment:
Maybe set a different `config` per chain to disambiguate more clearly b/w them, e.g "a", "b", "c":

```suggestion
	chainAConf := setupConfigInfo(chainA, p2pIDS, fChainA, []byte("chainA"))
	chainBConf := setupConfigInfo(chainB, p2pIDS[1:], fChainB, []byte("chainB"))
	chainCConf := setupConfigInfo(chainC, p2pIDS[2:], fChainC, []byte("chainC"))
```